stock_move_negative_stock_location/models/stock_move.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models
+from odoo.tools.float_utils import float_compare
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_confirm(self, merge=True, merge_into=False):
+        """Override to set the location_id of negative return moves to the default
+        location_dest_id of the return picking type."""
+        neg_r_moves = self.filtered(
+            lambda move: float_compare(
+                move.product_uom_qty, 0, precision_rounding=move.product_uom.rounding
+            )
+            < 0
+        )
+        self = self.with_context(neg_r_moves=neg_r_moves.ids)
+        return super()._action_confirm(merge=merge, merge_into=merge_into)
+
+    def set_negative_return_moves_location(self):
+        """Set the location_id of negative return moves to the default location_dest_id
+        of the return picking type."""
+        neg_r_moves = self.browse(self.env.context["neg_r_moves"])
+        for move in neg_r_moves:
```

Comment:
Better to get methods not depending on context but on current recordset, easier to test and understand:
```suggestion
        for move in self:
```